dashboard/app/controllers/courses_controller.rb

```diff
@@ -125,36 +87,28 @@ def edit
   end
 
   def vocab
-    # Assumes if one unit in a unit group is migrated they all are
-    return render :forbidden unless @unit_group.default_units[0].is_migrated
+    return render :forbidden unless can?(:vocab, @unit_group)
     @course_summary = @unit_group.summarize_for_rollup(@current_user)
   end
 
   def resources
-    # Assumes if one unit in a unit group is migrated they all are
-    return render :forbidden unless @unit_group.default_units[0].is_migrated
+    return render :forbidden unless can?(:resources, @unit_group)
     @course_summary = @unit_group.summarize_for_rollup(@current_user)
   end
 
   def code
-    # Assumes if one unit in a unit group is migrated they all are
-    return render :forbidden unless @unit_group.default_units[0].is_migrated
+    return render :forbidden unless can?(:code, @unit_group)
     @course_summary = @unit_group.summarize_for_rollup(@current_user)
   end
 
   def standards
-    if !@unit_group.present? && UnitGroup.family_names.include?(params[:course_name])
-      redirect_to_course = UnitGroup.latest_stable(params[:course_name])
-      redirect_to standards_course_path(redirect_to_course)
-      return
-    end
-    raise ActiveRecord::RecordNotFound unless @unit_group
-    # Assumes if one unit in a unit group is migrated they all are
-    return render :forbidden unless @unit_group.default_units[0].is_migrated
+    return render :forbidden unless can?(:standards, @unit_group)
```

Comment:
in general, you shouldn't have to do `return render :forbidden unless can?`, since this is pretty much exactly what `authorize_resource` does. can you see if everything still works without these calls?

Reply:
Ya my goal was to remove these but the tests would not pass if I did and I could not figure out why. Probably worth revisiting maybe with another set of eyes